sky/data/storage_utils.py

```diff
@@ -279,12 +279,18 @@ def _store_symlink(zipf, path: str, is_dir: bool):
                     zipf.write(item)
                 elif os.path.isdir(item):
                     for root, dirs, files in os.walk(item, followlinks=False):
+                        # Filter out directories in the excluded_files list
+                        # This prevents os.walk from traversing into excluded
+                        # directories
```

Comment:
```suggestion
                        # Modify dirs in-place to control os.walk()'s traversal behavior.
                        # This filters out excluded directories BEFORE os.walk() visits them,
                        # preventing traversal into any excluded directory and its contents.
                        # Note: dirs[:] = ... is required for in-place modification;
                        # a simple assignment dirs = ... would not affect the traversal.
```

Maybe we can elaborate more to help developers that are not familiar with `os.walk` like me 😉

Reply:
Nice! Thanks for the modifcation of the comment!